hack/verify-govulncheck.sh

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+# Copyright 2023 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -euo pipefail
+
+export WORKDIR=${ARTIFACTS:-$TMPDIR}
+export PATH=$PATH:$GOPATH/bin
+mkdir -p "${WORKDIR}"
+go install golang.org/x/vuln/cmd/govulncheck@v1.0.1
+
+govulncheck -scan module ./... > "${WORKDIR}/head.txt"
+
+git reset --hard HEAD
```

Comment:
we definitely don't want to do a hard reset like this... this would throw away anything uncommitted (!)

making the temporary worktree above lets us get the branch we're comparing against without disturbing their local worktree:

```
govulncheck -scan module ./... > "${KUBE_TEMP}/head.txt"
pushd "${WORKTREE}" >/dev/null
  govulncheck -scan module ./... > "${KUBE_TEMP}/pr-base.txt"
popd >/dev/null
```